src/client.ts

```diff
@@ -3117,9 +3123,10 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
             // Cache the key, if possible.
             // This is async.
             this.crypto.storeSessionBackupPrivateKey(privKey)
+                .then(cacheCompleteCallback)
                 .catch((e) => {
                     logger.warn("Error caching session backup key:", e);
-                }).then(cacheCompleteCallback);
+                });
```

Comment:
This changes the behaviour, the logger warn on L3128 will fire if the cacheCompleteCallback throws

Reply:
I believe the subsequent commit addresses this.I believe the subsequent commit addresses this.